src/main/java/com/amazon/ion/bytecode/bin10/ByteArrayBytecodeGenerator10.kt

```diff
@@ -2,14 +2,417 @@
+            }
+            else -> throw IonException("Encountered an illegal typeId; not a valid float length: $typeId")
+        }
+    }
+
+    private fun compileSymbolValue(source: ByteArray, position: Int, length: Int, dest: BytecodeBuffer) {
+        val sid = readUInt(source, position, length).toInt()
+        dest.add(Instructions.I_SYMBOL_SID.packInstructionData(sid))
+    }
+
+    private fun compileList(source: ByteArray, position: Int, length: Int, dest: BytecodeBuffer, cp: AppendableConstantPoolView) {
+        compileContainer(Instructions.I_LIST_START, dest) {
+            var p = position
+            val end = position + length
+            while (p < end) p += compileChildValue(source, p, dest, cp)
+        }
+    }
+
+    private fun compileSExp(source: ByteArray, position: Int, length: Int, dest: BytecodeBuffer, cp: AppendableConstantPoolView) {
+        compileContainer(Instructions.I_SEXP_START, dest) {
+            var p = position
+            val end = position + length
+            while (p < end) p += compileChildValue(source, p, dest, cp)
+        }
+    }
+
+    private fun compileStruct(source: ByteArray, position: Int, length: Int, dest: BytecodeBuffer, cp: AppendableConstantPoolView) {
+        compileContainer(Instructions.I_STRUCT_START, dest) {
+            var p = position
+            val end = position + length
+            while (p < end) {
+                val sidValueAndLength = VarIntHelper.readVarUIntValueAndLength(source, p)
+                val sid = sidValueAndLength.ushr(8).toInt()
+                p += sidValueAndLength.and(0xFF).toInt()
+                dest.add(Instructions.I_FIELD_NAME_SID.or(sid))
+                p += compileChildValue(source, p, dest, cp)
+            }
+        }
+    }
+
+    private inline fun compileContainer(instruction: Int, dest: BytecodeBuffer, content: () -> Unit) {
+        val containerStartIndex = dest.reserve()
+        val start = containerStartIndex + 1
+        content()
+        dest.add(Instructions.I_END_CONTAINER)
+        val end = dest.size()
+        dest[containerStartIndex] = instruction.packInstructionData(end - start)
+    }
+
+    private fun compileChildValue(source: ByteArray, position: Int, dest: BytecodeBuffer, cp: AppendableConstantPoolView, isAnnotated: Boolean = false): Int {
+        var p = position
+
+        val typeId = source[p++].unsignedToInt()
+
+        // Length counts only the bytes after the typeId.
+        val valueAndLength = getLengthForTypeId(typeId, source, p)
+        p += valueAndLength.toByte()
+        val length = valueAndLength.ushr(8).toInt()
+
+        if (TypeIdHelper.isNull(typeId)) {
+            compileNullValue(typeId, dest)
+        } else when (TypeIdHelper.operationKindForTypeId(typeId)) {
+            OperationKind.UNSET -> if (isAnnotated) throw IonException("Invalid annotation wrapper: NOP pad may not occur inside an annotation wrapper.")
```

Comment:
Is the relevant error message to all situations where this can occur?

Reply:
I believe it is relevant any time you have an NOP for which `isAnnotated` is true. Is there some other case that I'm missing here?

Reply:
I guess my question was about whether UNSET is only used for NOP. I think you're saying that it is, and therefore the error message is correct.